### ISSUE TITLE
New Rule: no-emphasis-as-heading

### ISSUE THREAD

**Issue by FloEdelmann** [eslint/markdown#683]
### Rule details

Prevent emphasized (i.e. bold or italic) text used to separate sections, where a heading should be used instead

### What type of rule is this?

Warns about a potential problem

### Example code

Bad:

```markdown
**My document**

Lorem ipsum dolor sit amet...

_Another section_

Consectetur adipiscing elit, sed do eiusmod.
```

Good:

```markdown
# My document

Lorem ipsum dolor sit amet...

## Another section

Consectetur adipiscing elit, sed do eiusmod.
```

### Prior Art

Markdownlint rule `MD036` (`no-emphasis-as-heading`): https://github.com/DavidAnson/markdownlint/blob/main/doc/md036.md

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by lumirlumir** [comment]
Thank you for opening an issue. Since `remark-lint` has the same rule, [`remark-lint-no-emphasis-as-heading`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-emphasis-as-heading#remark-lint-no-emphasis-as-heading), and we’ve generally accepted rules with prior art, I think this suggestion makes sense. It’s also not a stylistic rule and is intended to improve accessibility.

I can work on it, and I’ll leave this open for a bit before marking it as accepted to gather more inputs.

**Comment by lumirlumir** [comment]
I see two agreements on this suggestion, but could I get one more before marking it as accepted? @eslint/eslint-team

**Comment by lumirlumir** [comment]
**TSC Summary**: This issue proposes adding a `no-emphasis-as-heading` rule to report standalone bold or italic text used in place of Markdown headings. Similar rules exist in `markdownlint` and `remark-lint`, and the rule would improve document structure and accessibility.

**TSC Question**: Should we accept this proposal and add the rule to `@eslint/markdown`, or leave this pattern unreported?
